### ISSUE TITLE
New maintainers

### ISSUE THREAD

**Issue by betatim** [scikit-optimize/scikit-optimize#847]
Hello all,

scikit-optimize has seen a lot of new contributions recently from @holgern. He contacted the current maintainers (@glouppe, @iaroslav-ai and @MechCoder) asking if he could help out with maintenance.

Welcome Holger!

The guidelines for [interacting](CONDUCT.md) and [contributing](CONTRIBUTING.md) remain the same.

A main task for maintaining this project is around growing the number of people who actively contribute to the maintenance of the project in order to keep the level of quality high. This is a hard task! Depending on what part of the history of the project you look at the past maintainers did a good or a bad job at this. I look forward to seeing how new maintainers tackle this challenge and keep growing the community.

As part of this I would be happy to see more people creating PRs and increasing their involvement so that new people can join Holger in maintaining this project.

**Comment by kernc** [comment]
I would like to nominate myself as a new maintainer. :raised_hand_with_fingers_splayed: 

It's a real shame if not wholly unacceptable to see a such a well-rounded drop-in library stall unmaintained, accumulating untriaged issues even to the point of [incompatibilities with its dependencies](https://github.com/scikit-optimize/scikit-optimize/issues/981) and [downgrades as the recommended course of action](https://github.com/scikit-optimize/scikit-optimize/issues/978#issuecomment-751972201)!

Having just recently evaluated several options in the Python hyperparameter optimization space, I found scikit-optimize the one to go with for one of my projects, so I have some direct interest in it remaining maintained and useful.

If you'll consider my application, please see examples of my [reasoning](https://github.com/scikit-optimize/scikit-optimize/pull/974), [work](https://github.com/scikit-optimize/scikit-optimize/pull/971), and [interaction](https://github.com/scikit-optimize/scikit-optimize/search?q=commenter%3Akernc&type=issues) already in the issue tracker, with more available obviously in other projects I work on.

As a new maintainer, I'd begin by:
* [ ] porting CI to GitHub Actions (can't have CI broken or taking hours to complete),
* [ ] triaging existing issues and PRs (assigning labels like you guys have till late 2017), _tidying up_ and hopefully making everyone again somewhat more interested in contributing.

Thanks for your consideration.

cc: @scikit-optimize, @betatim, @glouppe, @iaroslav-ai, @MechCoder, @holgern

**Comment by betatim** [comment]
Hi @kernc 👋 

thanks for taking the time to find, evaluate and then start helping out here!

It would be great to meet you to chat a bit. For me a large part of maintaining a project is about the social side of things. Not just technical skills. How could we make this happen given everyone has a busy life, timezones, etc?

**Comment by kernc** [comment]
I got an interview! :raised_hands: Sure thing. It shouldn't be too hard to sync — we're all keeping busy, but at least we're mostly locked in. :laughing: Let me send you an email.

**Comment by xmatthias** [comment]
@betatim @kernc @holgern (@glouppe, @iaroslav-ai and @MechCoder) 

sorry for the ping - but i'd like to bring to your attention that scikit-optimize (this project) is (once again) in a pretty desolate state.
There's several breaking compatibility issues with recent numpy / scikit-learn versions  (#1171, #1137) - both of which would have fixing PR's available (#1187, #1184).

Unfortunately, CI is also failing due to the image used having reached end of life.
There's also contributions available (#1074) which would probably be fixing this (and seem very close to be working, if it's not good already).

I think the parts (community contributions, community interest) are available for this project to succeed - but it'll need some love for sure.

**Comment by betatim** [comment]
What way forward would you propose @xmatthias? I think from the side of the people you pinged no one has the time to either actively work on this project or find new maintainer(s). Which makes it difficult to see a way forward.

**Comment by xmatthias** [comment]
Well that's the problematic part, really.
if I'd have a solution, i'd have proposed that above.

Please treat the below as a rough set of ideas / my opinions only.

---

In my (naiive) understanding, scikit-optimize was part of the scikit-learn foundation / ecosystem - which might might make it eligible to either funding, or time-based contributions by scikit-learn funded contributors to at least keep basic compatibility (no new features ...).
But maybe / probably i'm wrong considering that this is a separate github org - so might be completely unrelated to scikit-learn (i might also be wrong about funding as part of scikit-learn - i'm not really familiar with this entity as an organization).

---

Ideally, an issue with "call for new maintainers" - which can then be vetted somewhat (you'll not want some random guy from the internet you never seen and never commited anything ... - maybe based on historic contributions to this / other similar packages, but criteria can essentially be anything) should be done.

Now based on your above comment, we doubt anyone will have time for this  - which essentially means that we have

* no active support to keep / restore compatibility
* no time to find successors

Which essentially declares the project dead - at least for now.

>  Which makes it difficult to see a way forward.

If that's the conclusion, then I think this should be clearly stated at the top of the project's readme, providing projects depending on scikit-optimize clarity on the project's future (or lack thereof) - which can aid / simplify their decision to either move to supported optimization frameworks, or to accept the drawbacks.

This doesn't mean scikit-optimize doesn't work as is, given the constraints on older scikit-learn and numpy versions - but it'll clarify expectations and most likely will also reduce frustration in issues - where right now, simply no answer is given.

With such a statement, it's clear that no answer should be expected (which won't mean there will be none - just that it's not very likely).
